File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 1 if len(nums) >= 1 else 0
        numSet = set(nums)
        for n in nums:
            cur = 1
            if n+1 in numSet and n-1 not in numSet:
                n2 = n
                while n2+1 in numSet:
                    cur += 1
                    longest = max(longest, cur)
                    n2 += 1
        return longest

File: Data Structures & Algorithms/integer-break/submission-3.rs
impl Solution {
    pub fn integer_break(n: i32) -> i32 {
        let mut dp = vec![1; n as usize];

        for num in 2..=n {
            dp[(num-1) as usize] = if num == n {0} else {num};
            for k in 1..num {
                let cur = dp[(num-1) as usize];
                let l = dp[(k-1) as usize];
                let r = dp[(num-k-1) as usize];
                dp[(num-1) as usize] = max(cur, l * r)
            }
        }

        dp[(n-1) as usize]
    }
}
